hazelcast/src/main/java/com/hazelcast/scheduledexecutor/impl/ScheduledExecutorDataSerializerHook.java

```diff
@@ -0,0 +1,136 @@
+    public static final int IS_DONE_OP = 13;
+    public static final int IS_CANCELED_OP = 14;
+    public static final int GET_STATS_OP = 15;
+    public static final int TASK_STATS = 16;
+
+    public static final int SYNC_STATE_OP = 17;
+    public static final int REPLICATION = 18;
+
+    public static final int DESTROY_TASK_OP = 19;
+
+    public static final int GET_ALL_SCHEDULED = 20;
+
+    public static final int SHUTDOWN = 21;
+
+    @Override
+    public int getFactoryId() {
+        return F_ID;
+    }
+
+    @Override
+    public DataSerializableFactory createFactory() {
+        return new DataSerializableFactory() {
+            @Override
+            public IdentifiedDataSerializable create(int typeId) {
+                switch (typeId) {
+                    case TASK_HANDLER:
+                        return new ScheduledTaskHandlerImpl();
+                    case BACKUP_DESCRIPTOR:
+                        return new BackupTaskDescriptor();
+                    case RUNNABLE_DEFINITION:
+                        return new TaskDefinition();
+                    case RUNNABLE_ADAPTER:
+                        return new ScheduledRunnableAdapter();
+                    case NAMED_TASK_DECORATOR:
+                        return new NamedTaskDecorator();
+                    case GET_DELAY_OP:
+                        return new GetDelayOperation();
+                    case CANCEL_OP:
+                        return new CancelTaskOperation();
+                    case CANCEL_BACKUP_OP:
+                        return new CancelTaskBackupOperation();
+                    case SCHEDULE_OP:
+                        return new ScheduleTaskOperation();
+                    case DESTROY_TASK_OP:
+                        return new DestroyTaskOperation();
+                    case IS_DONE_OP:
+                        return new IsDoneOperation();
+                    case IS_CANCELED_OP:
+                        return new IsCanceledOperation();
+                    case TASK_STATS:
+                        return new ScheduledTaskStatisticsImpl();
+                    case GET_STATS_OP:
+                        return new GetStatisticsOperation();
+                    case SCHEDULE_BACKUP_OP:
+                        return new ScheduleTaskBackupOperation();
+                    case SYNC_STATE_OP:
+                        return new SyncStateOperation();
+                    case REPLICATION:
+                        return new ReplicationOperation();
+                    case GET_ALL_SCHEDULED:
+                        return new GetAllScheduledOperation();
+                    case GET_RESULT:
+                        return new GetResultOperation();
+                    case PUBLISH_RESULT:
+                        return new ResultReadyNotifyOperation();
+                    case SHUTDOWN:
+                        return new ShutdownOperation();
+                    default:
+                        return null;
```

Comment:
throwing exception if `typeId` is not known would be better, `IllegalArgumentException` maybe ?